generated/nifake/tests/test_session.py

```diff
@@ -351,6 +345,25 @@ def test_error_get_vi_real64_attribute(self):
                 assert e.code == test_error_code
                 assert e.description == test_error_desc
 
+    def test_get_channel_based_attribute(self):
```

Comment:
In this test you get, in the context manager test, you set.

I think you should test the other options as well: set without context manager and get with it.

Reply:
I will expand on these tests